### PR TITLE
CONSOLE-4014: Pass OCM organization ID and OCP cluster ID to console-config.yaml

### PR DESCRIPTION
**Pull request by jhadvig**
Updating console-operator to get:

- `clusterID` - from the clusterversion `version` resource
- `organizationID` - from the OCM accountmanager API by passing clusterID and pull-secret for `cloud.openshift.com`

Refactored the sync_400 to have a single `GetTelemeterConfiguration` metyod which will be responsible for assembling the telemetry configuration.

/assign @TheRealJon 

@jerolimov FYI

### CODE REVIEW

File: pkg/console/subresource/configmap/configmap.go
Comment (TheRealJon) [inline pkg/console/subresource/configmap/configmap.go:189]: Did we lose this logic?

**Review by TheRealJon — CHANGES_REQUESTED** [review]

**Comment by jhadvig** [comment]
QE Approver:
/assign @yapei
Docs Approver:
/assign @opayne1
PX Approver:
/assign @RickJWagner

**Comment by jerolimov** [comment]
I got this errors on a cluster bot when enabling monitoring in the cluster-monitoring-config ConfigMap:

https://console/k8s/ns/openshift-monitoring/configmaps/cluster-monitoring-config 

And applying some anntations:

```
E0423 07:29:52.732198 1 status.go:130] ConfigMapSyncDegraded FailedGetTelemetryConfig HTTP request failed with status '403 Forbidden'
E0423 07:29:52.732384 1 base_controller.go:268] ConsoleOperator reconciliation failed: HTTP request failed with status '403 Forbidden'
E0423 07:29:55.008112 1 status.go:130] ConfigMapSyncDegraded FailedGetTelemetryConfig HTTP request failed with status '403 Forbidden'
E0423 07:29:55.008317 1 base_controller.go:268] ConsoleOperator reconciliation failed: HTTP request failed with status '403 Forbidden'
E0423 07:29:57.975301 1 status.go:130] ConfigMapSyncDegraded FailedGetTelemetryConfig HTTP request failed with status '403 Forbidden'
E0423 07:29:57.975600 1 base_controller.go:268] ConsoleOperator reconciliation failed: HTTP request failed with status '403 Forbidden'
```

/hold

File: pkg/console/telemetry/telemetry.go
Comment (jerolimov) [inline pkg/console/telemetry/telemetry.go:102]: Should we use the http proxy configuration here to ensure that it's working on airgapped clusters?

```suggestion
	client := &http.Client{
		Timeout: 30 * time.Second,
		Transport: &http.Transport{
			Proxy: http.ProxyFromEnvironment,
		},
	}
```

**Review by jerolimov — CHANGES_REQUESTED** [review]

**Comment by jhadvig** [comment]
> I got this errors on a cluster bot when enabling monitoring in the cluster-monitoring-config ConfigMap:
> 
> https://console/k8s/ns/openshift-monitoring/configmaps/cluster-monitoring-config
> 
> And applying some anntations:
> 
> ```
> E0423 07:29:52.732198 1 status.go:130] ConfigMapSyncDegraded FailedGetTelemetryConfig HTTP request failed with status '403 Forbidden'
> E0423 07:29:52.732384 1 base_controller.go:268] ConsoleOperator reconciliation failed: HTTP request failed with status '403 Forbidden'
> E0423 07:29:55.008112 1 status.go:130] ConfigMapSyncDegraded FailedGetTelemetryConfig HTTP request failed with status '403 Forbidden'
> E0423 07:29:55.008317 1 base_controller.go:268] ConsoleOperator reconciliation failed: HTTP request failed with status '403 Forbidden'
> E0423 07:29:57.975301 1 status.go:130] ConfigMapSyncDegraded FailedGetTelemetryConfig HTTP request failed with status '403 Forbidden'
> E0423 07:29:57.975600 1 base_controller.go:268] ConsoleOperator reconciliation failed: HTTP request failed with status '403 Forbidden'
> ```
> 
> /hold

Based on the story we dont wanna gather telemetry from CI clusters, so enabling the telemetry on a CI cluster will have it consequences, cause its not supported case.
Im not against handling this case, but if anything lets do it as a followup.

/hold cancel

**Comment by yapei** [comment]
verified that `CLUSTER_ID` and `ORGANIZATION_ID` can be correctly set in `cm/console-config` and `window.SERVER_FLAGS.telemetry`
/label qe-approved

**Review by jerolimov — APPROVED** [review]

**Comment by jhadvig** [comment]
flake
```
 ✖  crd-extensions/console-yaml-sample.      01:29        1        -        1        -        - │
  │    cy.ts  
```
/retest

**Pull request merged**
